server/src/main/java/org/eclipse/lsp/cobol/core/visitor/VisitorHelper.java

```diff
@@ -249,14 +251,49 @@ List<Node> createTreeNode(
         .map(constructNode(nodeConstructor, children))
         .orElse(children);
   }
+
   /**
-   * Retrieve a locality from the given mapping
-   *
-   * @param positions map of localities
-   * @param childToken token to retrieve
-   * @return optional of locality
+   * Gets a value from DataOccursClauseContext context
+   * @param ctx a context object
+   * @return extracted value
    */
-  Optional<Locality> getLocality(Map<Token, Locality> positions, Token childToken) {
-    return ofNullable(positions.get(childToken));
+  public Integer retrieveOccursToValue(CobolParser.DataOccursClauseContext ctx) {
```

Comment:
It will be better to return `Optional<Integer>` here. I understand that you move it here from CobolVisitor, but it was a private function before, and now it's public.

Reply:
Fixed